### ISSUE TITLE
Where can I just pick the plugin?

### ISSUE THREAD

**Issue by danielo515** [tobibeer/tw5-plugins#2]
Hello Tobias,

As I said before, I want to include your plugin library as part of Noteself default edition. I only need the plugin, not the library. Where can I pick it up ? I would prefer to pick it from a dist directory here on your repository, or clone your repository and build it. However, seems that the plugin is tied to the build process of the library itself. Any chance to separate them? Any other idea about this?

Regards

**Comment by tobibeer** [comment]
Each plugin has and is built in its own repo. The "library" simply is another repo that puts them all together in the library format and makes them available as such.

If you want to pre-bake my plugins into your distro, I recommend you clone the repos into your filesystem and then build your repo from there. Otherwise, if it's for your end-users to do, just make the library available.

I guess you'd always prefer to pull updates in batch and that's the nature of your request. ;-)

So, you possibly want *all* plugin sources in one repo for you to just include upon built. Will have to see if I can organize some sensible workflow around that. Not sure, possibly I should also update the library every time I update a plugin? Not sure I always did that in the past. Will need to check. Ideally though, whenever I push a new plugin release, I would hope to see that reflected in both the library as well as any not-yet--available "dist" folder.

Some help & pointers regarding how to automate this would be appreciated. Right now I use GULP to built, so perhaps I could have another script in a parent folder which then triggers building the library and dist as well. @felixhayashi?

**Comment by danielo515** [comment]
Hello @tobibeer 
I think you misunderstood my request. 
I don't want to pre-bake all your plugins in my distro, I want to pre-bake your plugin library, so when the users open the plugin library on their wikis they can see your plugin library, referencing your plugins from their respective sources.

I know that each of your plugins has a dist folder, which will be very convenient for anyone that want to pre-install them on any automatic process, but this is not my case. I'm only interested (currently) to pre-bake one of your plugins: your plugin library, which is in turn a plugin.

One plugin as a gate to all your plugins....

Regards

**Comment by tobibeer** [comment]
> so when the users open the plugin library on their wikis they can see your plugin library, referencing your plugins from their respective sources.

It appears, all you need to make available is this tiddler and it never needs any updating:
 [$:/pluginlibrary/tobibeer/official](http://tobibeer.github.io/tw5-plugins/#%24%3A%2Fpluginlibrary%2Ftobibeer%2Fofficial)

Not?

**Comment by danielo515** [comment]
Yes, that is the only tiddler that I need, which is in turn a plugin. You can "build" that tiddler as a JSON file and add it to your dist folder on your github repository just like @felixhayashi does for tiddlymap. That will make very easy to pick it programmatically on every build of my edition ensuring that I'm always up to date.
I'm not sure that it will never need any updating.

**Comment by tobibeer** [comment]
But, you see... seing as how this never needs updating, what repo does it need other than the one it is at? I truly don't get it. ;-)

Perhaps you tell me a bit about your workflow so I better understand at what point it is not enough to just incorporate that tiddler as you see it right there.

You could simply just clone the repo and include the tiddler in your build. (Or is it not possible to be selective about includes from other wiki / repo / plugin folders? tbh, I never actually tried)

So literally, all you want is a plugin-library plugin that you can then bake into your solution.

Slowly it's dawning on me. :D

I simply didn't think that this truly requires a plugin in order to distribute a single tiddler.
